src/components/AppConfig/Vector.tsx

```diff
@@ -57,66 +63,60 @@ export function VectorConfig({
   secretsSet,
   onChange,
   onChangeSecrets,
-}: {
-  settings?: AppPluginSettings;
-  secrets: Secrets;
-  secretsSet: SecretsSet;
-  onChange: (settings: VectorSettings) => void;
-  onChangeSecrets: (secrets: Secrets) => void;
-}) {
+}: Props<VectorSettings> & SecretProps) {
   // update the model value if the embedder type changes
   React.useEffect(() => {
-    if (settings?.vector?.embed?.type === 'openai' && settings?.vector?.model !== OPENAI_DEFAULT_MODEL) {
-      onChange({ ...settings.vector, model: OPENAI_DEFAULT_MODEL });
+    if (settings?.embed?.type === 'openai' && settings?.model !== OPENAI_DEFAULT_MODEL) {
+      onChange({ ...settings, model: OPENAI_DEFAULT_MODEL });
     } else if (
-      settings?.vector?.embed?.type === 'grafana/vectorapi' &&
-      (settings?.vector?.model === undefined || settings?.vector?.model === OPENAI_DEFAULT_MODEL)
+      settings?.embed?.type === 'grafana/vectorapi' &&
+      (settings?.model === undefined || settings?.model === OPENAI_DEFAULT_MODEL)
     ) {
-      onChange({ ...settings.vector, model: VECTORAPI_DEFAULT_MODEL });
+      onChange({ ...settings, model: VECTORAPI_DEFAULT_MODEL });
     }
-  }, [settings?.vector, settings?.vector?.embed?.type, onChange]);
+  }, [settings, settings?.embed?.type, onChange]);
 
   return (
     <FieldSet label="Vector Settings">
       <Field label="Enabled" description="Enable vector database powered features.">
         <Checkbox
           name="enabled"
           data-testid={testIds.appConfig.vectorEnabled}
-          value={settings?.vector?.enabled || false}
+          value={settings?.enabled || false}
           onChange={(e) => onChange({ ...settings, enabled: e.currentTarget.checked })}
```

Comment:
Oh no wait, it would have set it in the `VectorSettings`, but would also have filled `VectorSettings` with all of the root settings due to the `...settings` spread. You can see this if you start from empty settings, click the checkbox, then immediately hit "Save & test": the jsonData returned in the /settings API call has recursive `vector` keys.